kdbg/winstack.cpp

```diff
@@ -305,6 +309,22 @@ void WinStack::slotExpandCollapse(int)
     }
 }
 
+void WinStack::slotFlavorChanged(const QString& flavor)
+{
+    Q_UNUSED(flavor);
+    //Don't reload without a reason
+    if( m_isX86 ){
+        reloadAllFiles();
+    }
+}
+
+void WinStack::slotTargetChanged(const QString& target)
+{
+    m_target = target;
+
+    QRegularExpression reg = QRegularExpression(QStringLiteral(R"***(86)***"));
+    m_isX86 = m_target.contains(reg);
```

Comment:
Again, we do not need this member variable. There is only one user, which can do the `indexOf` call itself.

Which makes me think: this class uses the target name only to decide whether to reload the assembly. Why does it have to know the target name? The only time when the reloading can happen is when the *This Program* settings dialog is closed. That would be the right time to request the source windows to reload their assembly if it was x86. No target name required here.

Reply:
I used to print the target name in a tooltip, explaining that this "target" etc.. You are right.